src/components/UserProfile/UserProfile.jsx

```diff
@@ -1234,7 +1258,7 @@ function UserProfile(props) {
                   role={requestorRole}
                   onUserVisibilitySwitch={onUserVisibilitySwitch}
                   isVisible={userProfile.isVisible}
-                  canEditVisibility={canEdit && !['Volunteer', 'Mentor'].includes(requestorRole)}
+                  canEditVisibility={canEditVisibility}
```

Comment:
It seems like the only place `canEditVisibility` is used in this file is to pass it to `TeamsTab`. Could check the `editVisability` permission in the `TeamsTab` component instead of passing it as a prop.

Reply:
I cannot find any line with canEditVisibility = {canEdit && !['Volunteer']...... in src/components/userProfile/userprofile.jsx. Please check again.

Reply:
The code comparison is because it's a comment on your change. Not a suggested change.

Reply:
Since, that variable canEditVisibility is used multiple times on the page, it doesn't make any sense to make any changes.